docs/topics/api-guidelines-minimizing-mental-complexity.md

```diff
@@ -0,0 +1,685 @@
+Use classes to represent data and state in your API. When the data and state is hierarchical, consider using inheritance.
+
+If all the state required can be passed as parameters, prefer using top-level functions.
+When calls to these functions will be chained, consider writing them as extension functions to improve readability.
+
+### Choose the appropriate error handling mechanism
+
+Kotlin provides several mechanisms for error handling.
+Your API can throw an exception, return a `null` value, use a custom result type, or use the built-in [`Result`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/) type. Ensure that your library uses these options consistently and appropriately.
+
+When data cannot be fetched or calculated, use a nullable return type and return `null` to indicate missing data.
+In other cases, throw an exception or return a `Result` type.
+
+Consider providing overloads of functions, where one throws an exception, while the other wraps it in a result type instead.
+In these cases, use the `Catching` suffix to indicate that exceptions are caught in the function.
+For example, the standard library has the [`run`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/run.html) and [`runCatching`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/run-catching.html) functions using this convention,
+and the coroutines library has [`receive`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-receive-channel/receive.html) and [`receiveCatching`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-receive-channel/receive-catching.html) methods for channels.
+
+Avoid using exceptions for normal control flow. Design your API to allow for condition checks before attempting operations, thus preventing unnecessary error handling.
+[Command / Query Separation](https://martinfowler.com/bliki/CommandQuerySeparation.html) is a useful pattern that can be applied here.
+
+### Maintain conventions and quality
+
+The final aspect of consistency relates, not to the design of the library itself, but to maintaining a high level of quality.
+
+You should use automated tools (linters) for static analysis to ensure your code follows both general Kotlin conventions
+and project-specific conventions.
+
+A Kotlin library should also provide a suite of unit and integration tests covering all documented behaviors of all the API entry points.
+Tests should include a wide range of inputs, especially known boundary and edge cases. Any untested behavior should be assumed to be (at best) unreliable.
+
+Use this suite of tests during development to verify that changes do not break existing behavior.
+Run these tests on every release as part of a standardized build and release pipeline.
+Tools like [Kover](https://github.com/Kotlin/kotlinx-kover) can be integrated into your build process to measure coverage and generate reports.
+
+## Predictability
```

Comment:
Why did you decide to leave it on one page?
Out of curiosity

Reply:
I mean Predictabiliy, Debuggability...

Reply:
as discussed - it's still open - let's revisit later! :)